csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs

```diff
@@ -170,12 +170,33 @@ public void AppendExecutionProvider_Nuphar(string settings = "")
         #endregion //ExecutionProviderAppends
 
         #region Public Methods
+
+        /// <summary>
+        /// (Deprecated) Loads a DLL named 'libraryPath' and looks for this entry point:
+        /// OrtStatus* RegisterCustomOps(OrtSessionOptions* options, const OrtApiBase* api);
+        /// It then passes in the provided session options to this function along with the api base.
+        /// Deprecated in favor of RegisterCustomOpLibraryV2() because it provides users with the library handle 
+        /// to release when all sessions relying on it are destroyed
+        /// </summary>
         public void RegisterCustomOpLibrary(string libraryPath)
         {
             IntPtr libraryHandle = IntPtr.Zero;
             NativeApiStatus.VerifySuccess(NativeMethods.OrtRegisterCustomOpsLibrary(handle, libraryPath, out libraryHandle));
         }
 
+        /// <summary>
+        /// Loads a DLL named 'libraryPath' and looks for this entry point:
+        /// OrtStatus* RegisterCustomOps(OrtSessionOptions* options, const OrtApiBase* api);
+        /// It then passes in the provided session options to this function along with the api base.
+        /// The handle to the loaded library is returned in 'libraryHandle'. 
+        /// It can be freed by the caller after all sessions using the passed in
+        /// session options are destroyed, or if an error occurs and it is non null.
+        /// </summary>
+        public void RegisterCustomOpLibraryV2(string libraryPath, out IntPtr libraryHandle)
```

Comment:
>libraryHandle [](start = 77, length = 13)

What does the customer do with it ? Do we have an API to unload?

Reply:
Can the customer P/Invoke FreeLibrary() in kernel.dll ? The test I included does this. Either way, we need to return a handle to the user for him to decide what to do with it (otherwise it just remains leaking)

Reply:
I this case, I meant the wording of the comment. freed -> unloaded

---
In reply to: [513738673](https://github.com/microsoft/onnxruntime/pull/5608#discussion_r513738673) [](ancestors = 513738673)

Reply:
Modified the comment